hw012/update_mailroom.py

```
@@ -0,0 +1,147 @@
+        if path == "q" or path == "quit":
+            quit()
+        elif path == "r":
+            generate_report()
+        elif path == "t":
+            thankyou()
+        else:
+            return unvalid
+    except TypeError:
+        return unvalid
+
+
+def thankyou():
+    menu_option = """
+    Please enter a name or choose from the following:\n
+    list - Print a list of previous donors
+    quit - Return to the main menu\n
+    Input: """
+
+    print(menu_option)
+
+    donor = input()
+    check_thankyou_path(donor)
+
+
+def check_thankyou_path(donor):
+
+    if donor.lower() == "quit":
+        start_screen()
+    elif donor.lower() == "list":
+        print("See below for a list of donor names")
+        print(donor_name_only())
+        thankyou()
+    try:
+        if donor_list[donor]:
+            donor_add(donor)
+    except KeyError:
+        donor_add(donor)
+
+
+def generate_report():
+    """Generates a report for user with key information
+       still doesn't print sorted"""
+    print("Name \t\t|Total \t|# | Average\n" + ("_" * 40))
+
+    sorted_list = sorted(donor_list, key=lambda donor: sum(donor_list[donor]),
+                         reverse=True)
+    for key in sorted_list:
+        donor_name = key
+        total_donated = sum(donor_list[key])
+        num_donate = len(donor_list[key])
+        avg = total_donated / num_donate
+        print('%s \t|$%d \t|%d | $%d' %
+              (donor_name, total_donated, num_donate, avg))
+    return False
+    start_screen()
+
+
+def donor_name_only():
+    """Prints list of donor names from dictionary."""
+    for key in donor_list:
+        print(key)
+
+
+def donor_add(donor):
+    """Add donor to database"""
+    if donor not in donor_list:
+        donor_list[donor] = []
+        donation = donation_add(donor)
+        while not isinstance(donation, float) or donation < 0:
+            donation = donation_add(donor)
+
+    donor_list[donor].append[donation]
```

This is causing the following error:
`TypeError: 'builtin_function_or_method' object is not subscriptable`
I think it's being caught in a `try` statement somewhere and silently passing.
What you want to do is `donor_list[donor].append(donation)`